### ISSUE TITLE
Notification system for completion of action (feature request)

### ISSUE THREAD

**Issue by Siraris** [elastic/elasticsearch#9645]
Not sure if this is a feature that already exists in Elasticsearch, but I'm finding it would be nice to have some sort of notification when certain actions complete.  For example, if I do a snapshot, I don't know of a way when the snapshot completes.  It just says "Snapshot initiated".  Or if I use the https://github.com/codelibs/elasticsearch-reindexing plugin, it just says that it acknowledged the reindex, but it doesn't notify me when it's complete.  Being able to hook in and say "Send an email, or an SMS when X is done" would be a nice reassurance that my action was complete.

**Comment by clintongormley** [comment]
Hi @Siraris 

Yes I agree - this is a nice idea, and something we would like to pursue.  Currently it is blocked by the lack of a task management API, but we are actively working on implementing this: https://github.com/elasticsearch/elasticsearch/issues/6914

**Comment by clintongormley** [comment]
Closing in favour of #15117
